webapp/src/Controller/Jury/TeamController.php

```diff
@@ -421,21 +422,33 @@ public function deleteAction(Request $request, int $teamId)
     public function addAction(Request $request): Response
     {
         $team = new Team();
-        $team->setAddUserForTeam(true);
-        $team->addUser(new User());
+        $team->setAddUserForTeam(Team::CREATE_NEW_USER);
         $form = $this->createForm(TeamType::class, $team);
 
         $form->handleRequest($request);
 
         if ($form->isSubmitted() && $form->isValid()) {
             /** @var User $user */
             $user = $team->getUsers()->first();
-            if (!$team->getAddUserForTeam()) {
-                // If we do not want to add a user, remove it again
-                $team->removeUser($user);
-            } else {
-                // Otherwise, set the user's name to the team name
+            if ($team->getAddUserForTeam() === Team::CREATE_NEW_USER) {
+                // Create a user for the team
+                $user = new User();
+                $user->setUsername($team->getAddUserForTeam());
+                $team->addUser($user);
+                // Make sure the user has the team role to make validation work
+                /** @var Role $role */
+                $role = $this->em->createQueryBuilder()
```

Comment:
Is this the best way we can retrieve a Role? Just wondering if we could make some use of some helper for that maybe.

Reply:
Lol I copied this from the previous place but it is indeed super ugly. Pushing a new version now.